### ISSUE TITLE
UnLazyImage component doesn't display new image after reactively changing :src prop

### ISSUE THREAD

**Issue by talaxasy** [johannschopplich/unlazy#52]
### Reproduction

Reproduction link: [link](https://codesandbox.io/p/devbox/unlazyimage-4vm696?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clx8yixqx00073o6jta4ti6fu%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clx8yixqw00023o6ju8elegdt%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clx8yixqw00043o6jdgx5h40k%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clx8yixqx00063o6j6lk85ywp%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clx8yixqw00023o6ju8elegdt%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx8yixqw00013o6j19zr7xoh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%257D%255D%252C%2522id%2522%253A%2522clx8yixqw00023o6ju8elegdt%2522%252C%2522activeTabId%2522%253A%2522clx8yixqw00013o6j19zr7xoh%2522%257D%252C%2522clx8yixqx00063o6j6lk85ywp%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx8yixqx00053o6j4656t5lc%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clx8yixqx00063o6j6lk85ywp%2522%252C%2522activeTabId%2522%253A%2522clx8yixqx00053o6j4656t5lc%2522%257D%252C%2522clx8yixqw00043o6jdgx5h40k%2522%253A%257B%2522id%2522%253A%2522clx8yixqw00043o6jdgx5h40k%2522%252C%2522activeTabId%2522%253A%2522clx8yixqw00033o6jucaqyt6l%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx8yixqw00033o6jucaqyt6l%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clx8yji0u00673o6ji0r1e01t%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clx8yjhxz000udjeifnlvhj0s%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

Insert any image from the ones provided, for example Ivan Grozny, and click on the "Update Image" button.
The image is expected to change.
In fact, what happens is that the image remains the same

### Describe the bug

After reactively changing the image URL, the UnLazyImage component does not render the new value inside the src prop.

### Additional context

![image](https://github.com/johannschopplich/unlazy/assets/25802675/8970795b-661a-4d50-9cca-6ae23cada83f)

**Comment by felixranesberger** [comment]
I looked into the source code and I think this is not implemented at the moment. 
You can probably use `:key="currentImageUrl"` to force a re-render of the whole component and its lazy loading functionality when the value changes.

**Comment by johannschopplich** [comment]
Great catch! Yeah, the reactiveness is missing at the moment… PR welcome!

**Comment by talaxasy** [comment]
Yes, I already use `:key` for rerendering, can we consider the issue closed?

**Comment by johannschopplich** [comment]
Please leave it open. Technically speaking, it's a bug and everyone rightfully can expect props to be reactive. 😌
